src/librustdoc/doctree.rs

```diff
@@ -29,6 +59,63 @@ impl Module<'hir> {
             foreigns: Vec::new(),
             macros: Vec::new(),
             is_crate: false,
+            from_glob: false,
+        }
+    }
+
+    pub(crate) fn push_item(&mut self, ctx: &DocContext<'_>, new_item: Item<'hir>) {
+        let new_item_ns = ctx.tcx.def_kind(new_item.hir_item.def_id).ns();
+        if !new_item.name().is_empty() && new_item_ns.is_some() {
```

Comment:
Is it possible for `new_item.name()` to be empty? And is it possible that `new_item_ns` is empty?

Reply:
> Is it possible for `new_item.name()` to be empty? And is it possible that `new_item_ns` is empty?

The only situation I found which `new_item.name().is_empty()` is `std::prelude` (but IMO we don't need to care about it), remove this check seems won't affect the test result. But [this comment](https://github.com/rust-lang/rust/pull/83872#discussion_r607036466) mentioned some possible situation.  @ollie27 can you give a more detailed example?

On the other hand, `new_item_ns` can be empty in many situations like macro derives (a minimal example is `Debug` in `src/test/rustdoc/wrapping.rs`). If we remove this check, then adding those items may cause panic on L88.

Reply:
`new_item.name` can be empty for impl blocks - or if it's not and it's using `{{impl}}` or something, that's another bug, because impl blocks shouldn't use globbing logic, they don't have have a namespace.

Reply:
So means in this case both checks are still necessary? Maybe we should leave a comment or it's not necessary?